hypothesis-python/tests/cover/test_stateful.py

```diff
@@ -1333,9 +1438,157 @@ def use_flatmap(self, bun):
             assert isinstance(bun, int)
             return bun
 
+        @rule(
+            target=buns,
+            bun=buns.flatmap(lambda x: just(-x)).filter(lambda x: x < -1),
+        )
+        def use_flatmap_filtered(self, bun):
+            assert isinstance(bun, int)
+            assert bun < -1
+            return -bun
+
+        @rule(
+            target=buns,
+            bun=buns.flatmap(lambda x: just(x + 1)).map(lambda x: -x),
+        )
+        def use_flatmap_mapped(self, bun):
+            assert isinstance(bun, int)
+            assert bun < 0
+            return -bun
+
+        @rule(bun=buns)
+        def use_directly(self, bun):
+            assert isinstance(bun, int)
+            assert bun >= 0
+
+    Machine.TestCase.settings = Settings(stateful_step_count=5, max_examples=10)
```

Comment:
I'd leave the default `max_examples=` here; because swarm testing (disabling a random subset of rules) makes coverage less reliable than you might think for small n.